src/librustc_mir/error_codes.rs

```diff
@@ -2424,6 +2424,35 @@ const fn foo() -> impl T {
 ```
 "##,
 
+E0724: r##"
+Support for Non-Lexical Lifetimes (NLL) has recently been added to the Rust
+compiler. This uncovered some bugs in the old borrow borrow checker which in
+some cases allowed invalid code to compile, resulting in memory safety issues.
+
+### What do I do?
+
+You should change your code so the warning is no longer trigger. For backwards
+compatibility, this invalid code may still be compiled right now. However, at
+some point in the future, the compiler will no longer accept this code and will
+throw a hard error.
+
+### Shouldn't you fix the old borrow checker?
+
+The old borrow checker has known soundness issues that are basically impossible
+to fix. The new NLL-based borrow checker is the fix.
+
+### Can I turn these warnings into errors by denying a lint?
+
+No.
+
+### When are these warnings going to turn into errors?
+
+FIXME
```

Comment:
FIXME

Reply:
would it suffice to say here that:
 1. no formal timeline for turning the warnings into errors has been set, 
 2. see issue #58781 for more information?